rust-mode.el

```diff
@@ -96,6 +96,13 @@
     )
    ))
 
+;;; Start of a Rust item
+(defvar rust-top-item-beg-re
+  (concat "\\s-*\\(?:priv\\|pub\\)?\\s-*"
```

Comment:
Ok, I was mistaken about this.  What was happening was that the first draft of my new test case included a blank line, but these aren't currently handled very well by the indentation tests, because those add a six space prefix to each line, which then isn't removed from an otherwise blank line when reindenting.

Still, I've added the new test and made one minor tweak, which is to move the \\_> into the regexp here.